stubs/CoreGenericFunctions.phpstub

```diff
@@ -1300,4 +1300,11 @@ function get_defined_constants(bool $categorize = false): array {}
  * @param class-string $class
  * @return ($allow_string is false ? ($object_or_class is object ? bool : false) : bool)
  */
-function is_a($object_or_class, string $class, $allow_string = false): bool{}
\ No newline at end of file
+function is_a($object_or_class, string $class, $allow_string = false): bool{}
+
+/**
+ * @template T of array|string
+ * @param T $string
+ * @return (T is array<int, string> ? array<int, string> : T is array ? array : string)
```

Comment:
we should add |false everywhere in the return types

Reply:
Ah, good catch. But only at the string return.

I still wonder what happens, when you add a string that triggers returning false for a string into an array…